src/components/Badge/Badge.tsx

```diff
@@ -1,61 +1,53 @@
 import * as React from 'react';
 import {oneOf, node} from 'prop-types';
-import {Badge as CoreBadge} from 'wix-ui-core/Badge';
-import {ThemedComponent} from 'wix-ui-theme';
-
-import {theme} from './theme';
-import {UIText} from '../UIText';
-import {SKIN, FORM, Form, Skin} from './constants';
-
-const iconStyles = {
-  prefix: {paddingRight: '8px'},
-  suffix: {paddingLeft: '8px'}
-};
-
-const createBadgeIcon = (type, icon, size) => (
-  icon ?
-    <span style={iconStyles[type]}>
-      {React.cloneElement(icon, {size})}
-    </span> :
-    null
-);
+import {Badge as CoreBadge, BadgeProps as CoreBadgeProps} from 'wix-ui-core/StylableBadge';
+import style from './Badge.st.css';
+import {withStylable} from 'wix-ui-core/withStylable';
+import {UIText} from '../StylableUIText';
+import {SKIN, TYPE, Type, Skin} from './constants';
 
 interface BadgeProps {
-  form?: Form;
+  type?: Type;
   skin?: Skin;
   prefixIcon?: any;
```

Comment:
I would like to have an Icon type imported from `wix-ui-icons-common`...

Reply:
@shlomitc @alex5il is that something that in progress by any chance? if not, I will merge with this type

Reply:
It's still in progress, so let's make this one merged